lib/frames.js

```diff
@@ -126,6 +126,38 @@ function defineFrame(type, definition) {
   return Frame;
 }
 
+// restricted type helpers
+function role(value) {
```

Comment:
yeah I mean, most of these are just "choices", so we could instead set `requires: { mapping: 'of', objects: 'here' }`, but I figured doing it with a function was very easy for now - and doesn't cost too much